### ISSUE TITLE
GitHub Action to Detect Unused Dependencies

### ISSUE THREAD

**Issue by palisadoes** [PalisadoesFoundation/talawa-admin#858]
**Describe the bug**
We need a way to automatically detect unused dependencies as part of our GitHub Actions and fail if any are found.

1. Receiving alerts for unused dependencies and performing upgrades to these dependencies is pointless. 
2. It also exposes our code base to potential exploits even though the risk is low

**To Reproduce**

N/A

**Expected behavior**

See Above

**Actual behavior**

See Above

**Screenshots**

N/A

**Additional details**

This package and other similar ones may be useful 

- https://github.com/depcheck/depcheck

**Comment by anshgoyalevil** [comment]
@palisadoes 

Would like to work on it. Please assign it to me.

**Comment by rishav-jha-mech** [comment]
I was working near this issue, I came across these two libraries which does the job
1. depcheck  - https://www.npmjs.com/package/depcheck (Palisadoes mentioned above)
2. npm-check - https://www.npmjs.com/package/npm-check

But they need to be properly configured otherwise, they will show the packages that are actually being used as unused.

**Comment by anshgoyalevil** [comment]
@rishav-jha-mech 

Thanks. I will take a look.

**Comment by anshgoyalevil** [comment]
@palisadoes 

Sir, I tried implementing this thing in Talawa API and Admin. I have some queries regarding this:

- There are many false positives. 
I tested different packages for checking the unused dependencies, but most of them are giving such packages as unused, which are being used.

There is an option in all these tools, which allow us to ignore the dependencies which we don't want to be shown as unused. The problem with that is:
- It is not future proof. Say for example, we ignore keep some package in ignore list, and later it is no longer used actually. Then, we won't get to know about that.
- Even if we go for adding the ignore parameter, so false positives doesn't come up, we have opt for either of these two:
    - Do we want workflow to fail, if unused packages are found?
    - Do we just want to get a console.log inside the Workflow runs? This basically is useless.


My suggestion:

- We should manually remove the un-used dependencies for the first time.
- There is already an ```unauthorized file check``` workflow, which prevents adding new dependencies which are not being used.
- If any existing dependency's code is removed, and it is no longer used, we can go for checking it while reviewing the code, or say in a month or two. Since, such cases occur very very less.

What's your suggestion?
